mantidproject/mantid
# Simulate resolution for Iqt Monte Carlo Error calculation

### Description of work
This PR fixes a bug where the error value for the first bin in the monte carlo error calculation was zero.

The sample is randomized within its y error values, and now the resolution is also randomized within its y error values. During normalisation, the sample is divided by the resolution. Previously, this would generate a y value in the first bin with value 1, and an error of 0.

In the latest changes, the y value in the first bin is still 1, but the error is now non-zero. It makes sense that the errors in the randomized sample and resolution counts should be propagated to the result of the normalisation.

To test this change, I have done some fitting on the I(Q, t) tab of QENS Fitting. The new fits have a much better chi squared now that the first bin has a non-zero error.

*There is no associated issue.*

### To test:
1. Open Inelastic->Data Processor interface
2. Load the `irs26176_graphite002_red` as sample from the sample data
3. Load the `irs26173_graphite002_res` as resolution from the sample data
4. Tick `Enforce Normalisation`.
5. Click Run. Wait. Plot the first spectrum of the result `_iqt` workspace with error bars:
![image](https://github.com/mantidproject/mantid/assets/40830825/95a1ec32-84eb-4479-b148-f7ad1f40d18e)
6. Notice the error for the first bin is non-zero
7. Untick `Enforce Normalisation`.
8. Click Run. Wait. Plot the first spectrum of the result `_iqt` workspace with error bars:
![image](https://github.com/mantidproject/mantid/assets/40830825/ef6ec1f1-6942-4440-8e14-9b09cf81ae95)
9. Tick `Enforce Normalisation`.
10. Click Run. 
11. Open Inelastic-> QENS Fitting
12. Load the `_iqt` workspace from the ADS
13. Select `Fit Type` to be Stretched Exp
14. Select `Background` to be `Flat Background`
15. Click Run and wait.
16. The Chi squared above the Function Browser should be roughly 1.28 . Previously this was higher


### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
